metals/src/main/scala/scala/meta/internal/metals/testProvider/frameworks/MunitTestFinder.scala

```diff
@@ -0,0 +1,169 @@
+package scala.meta.internal.metals.testProvider.frameworks
+
+import scala.annotation.tailrec
+import scala.collection.immutable
+import scala.collection.mutable
+
+import scala.meta.Defn
+import scala.meta.Lit
+import scala.meta.Pkg
+import scala.meta.Term
+import scala.meta.Tree
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.testProvider.FullyQualifiedName
+import scala.meta.internal.metals.testProvider.TestCaseEntry
+import scala.meta.internal.parsing.Trees
+import scala.meta.internal.semanticdb.SymbolOccurrence
+import scala.meta.internal.semanticdb.TextDocument
+import scala.meta.io.AbsolutePath
+
+class MunitTestFinder(trees: Trees) {
+
+  def findTests(
+      doc: TextDocument,
+      path: AbsolutePath,
+      suiteName: FullyQualifiedName
+  ): Vector[TestCaseEntry] = {
+    val uri = path.toURI
+    val testcases = new mutable.ArrayBuffer[TestCaseEntry]()
+    val occurences = doc.occurrences
+      .filter(_.symbol.startsWith("munit/FunSuite#test"))
```

Comment:
`symbol.startsWith("munit/FunSuite#test")` won't work for
```scala
abstract class BaseMunitSuite extends munit.FunSuite {
}

class MunitTestSuite extends BaseMunitSuite {
  test("") {}
}
```
because now test symbol looks like `munit/BaseFunSuite#test()`.
I can check for munit/ on start and #test() at the end or I can get info about parent classes.

Reply:
We should be able to find the SymbolInformation for that definition and check if that symbol overrides `munit/FunSuite#test`